frontend/src/old-pages/Clusters/Instances.tsx

```diff
@@ -42,77 +42,94 @@ import {InstanceStatusIndicator} from '../../components/Status'
     clusterName &&
-      GetClusterInstances(clusterName, () =>
-        clearState(['app', 'clusters', 'action', 'pending']),
-      )
+      GetClusterInstances(clusterName, () => {
+        resetPending
+      })
   }
 
-  const stopInstance = (instance: Instance) => {
-    setState(['app', 'clusters', 'action', 'pending'], true)
-    Ec2Action([instance.instanceId], 'stop_instances', refresh)
-  }
+  const stopInstance = React.useCallback(
+    (instance: Instance) => {
+      setStopPending(true)
+      Ec2Action([instance.instanceId], 'stop_instances', refresh)
+    },
+    [instance],
+  )
 
-  const startInstance = (instance: Instance) => {
-    setState(['app', 'clusters', 'action', 'pending'], true)
-    Ec2Action([instance.instanceId], 'start_instances', refresh)
-  }
+  const startInstance = React.useCallback(
+    (instance: Instance) => {
+      setStartPending(true)
+      Ec2Action([instance.instanceId], 'start_instances', refresh)
+    },
+    [instance],
+  )
+
+  const isComputeFleetStopped = React.useMemo(
+    () => fleetStatus === ComputeFleetStatus.Stopped,
+    [fleetStatus],
+  )
+  const isHeadNodeRunning = React.useMemo(
+    () =>
+      instance &&
+      instance.nodeType === NodeType.HeadNode &&
+      instance.state === InstanceState.Running,
+    [instance],
+  )
+  const isHeadNodeStopped = React.useMemo(
+    () =>
+      instance &&
+      instance.nodeType === NodeType.HeadNode &&
+      instance.state === InstanceState.Stopped,
+    [instance],
+  )
 
   return (
     <SpaceBetween direction="horizontal" size="s">
-      {fleetStatus === ComputeFleetStatus.Stopped && (
-        <div>
-          {instance.nodeType === NodeType.HeadNode &&
-            instance.state === InstanceState.Running && (
-              <Button
-                loading={pending}
-                onClick={() => {
-                  stopInstance(instance)
-                }}
-              >
-                {t('cluster.instances.actions.stop')}
-              </Button>
-            )}
-          {instance.nodeType === NodeType.HeadNode &&
-            instance.state === InstanceState.Stopped && (
-              <Button
-                loading={pending}
-                onClick={() => {
-                  startInstance(instance)
-                }}
-              >
-                {t('cluster.instances.actions.start')}
-              </Button>
-            )}
-        </div>
-      )}
-      {fleetStatus !== ComputeFleetStatus.Stopped && (
-        <div title={t('cluster.instances.computeFleetInfo')}>
-          {instance.nodeType === NodeType.HeadNode &&
-            instance.state === InstanceState.Running && (
-              <Button disabled={true}>
-                {t('cluster.instances.actions.stop')}
-              </Button>
-            )}
-        </div>
-      )}
+      <Button
+        disabled={!(instance && isComputeFleetStopped && isHeadNodeRunning)}
+        loading={stopPending}
+        onClick={() => {
```

Comment:
suggestion: if you want to make use of your `useCallback`s, you should either memo-ize this function as well, or pass the `stopInstance` directly (same goes for `startInstance`)